src/React/Resources/assets/src/register_controller.ts

```diff
@@ -9,8 +9,20 @@
 
 'use strict';
 
+import { ComponentClass, FunctionComponent } from 'react';
+
+type Component = string | FunctionComponent<object> | ComponentClass<object, any>;
+
+declare global {
```

Comment:
This piece of code is inspired from a StackOverflow example and the Vue example

Reply:
The only part I'm not sure about is how you define the `type Component`. But, I think it's pretty safe to use this and if there is a better way, someone will tell us :). This is just to make TypeScript happier. So, I think the worst-case scenario is that someone tells us a nicer way to make TypeScript happier.

Reply:
I agree with you. I just looked at how Typescript inferred the type where it is used.

If someone knows a better way, let's use it